extensions/amp-bind/amp-bind.md

```diff
@@ -164,38 +182,105 @@ This state can be initialized with the `amp-state` component:
+[Expressions](#expressions) can reference state variables via dot syntax. In this example, `myState.foo` will evaluate to `"bar"`.
+
+Note that an `<amp-state>` element's child JSON has a maximum size of 100KB. 
 
-#### `AMP.setState()`
+`<amp-state>` can also specify a CORS URL instead of a child JSON script. See the [Appendix](#amp-state-specification) for details.
 
-State can be mutated by the [`AMP.setState()` action](../../spec/amp-actions-and-events.md).
+#### Updating state with `AMP.setState()`
 
-- `AMP.setState()` performs a deep merge of its arguments with the document state up to a depth of 10.
-- `AMP.setState()` can override variables initialized by `amp-state`.
+The [`AMP.setState()` action](../../spec/amp-actions-and-events.md) merges an object literal into the state. For example:
+
+```html
+<!-- Like JavaScript, you can reference existing
+variables in the values of the state update object literal. -->
+<button on="tap:AMP.setState({foo: 'bar', baz: myAmpState.someVariable})"></button>
+```
+
+When this button is pressed, `AMP.setState()` will [deep-merge](#deep-merge-with-ampsetstate) the update with the state up to a depth of 10. All variables, including those introduced by `amp-state`, can be overidden.
+
+When triggered by certain events, `AMP.setState()` also can access event-related data on the `event` property. 
+
+```html
+<!-- The "change" event of this <input> element
+contains a "value" variable that can be referenced via "event.value". -->
+<input type="range" on="change:AMP.setState({myRangeValue: event.value})">
+```
+
+
+ See [Actions and Events in AMP](../../spec/amp-actions-and-events.md) for more details.
+
+### Expressions
+
+Expressions are similar to JavaScript with some important differences.
+
+#### Differences from JavaScript
+
+- Expressions may only access the containing document's [state](#state).
+- Expressions **do not** have access to globals like `window` or `document`.
+- Only [whitelisted functions](#whitelisted-functions) are allowed.
+- Custom functions, classes and some control flow statements (e.g. `for`) are disallowed.
+- Undefined variables and array-index-out-of-bounds return `null` instead of `undefined` or throwing errors.
+- A single expression is currently capped at 50 operands for performance reasons. Please [contact us](https://github.com/ampproject/amphtml/issues/new) if this is insufficient for your use case.
+
+The full expression grammar and implementation can be found in [bind-expr-impl.jison](./0.1/bind-expr-impl.jison) and [bind-expression.js](./0.1/bind-expression.js).
+
+#### Examples
+
+The following are all valid expressions.
+
+```javascript
+// 'Hello World 3'
+'Hello ' + 'World ' + 3
+// 2209
+47 * +'47'
+// Boolean value of event.checked
+event.checked && (true || !!1)
+// 'Deal' if `event.value` is greater than
+// the median of the range, else 'No deal'
+event.value > ((event.max - event.min) / 2) ? 'Deal' : 'No deal'
```

Comment:
I think we should provide simple, practical examples. I can't imagine `(true || !!1)` being used in the real world.